### DISCUSSION TITLE
Jaeger and Istio Sidecar Compatibility

### DISCUSSION THREAD

**Discussion by AnthonyGurg** [jaegertracing/jaeger-operator#1373] [Q&A]
Hi there, I had a question about the sidecar compatibilities.

I'm currently using the Red Hat OSSM 2.0 and OCP 4.6 and I'm normally able to view my traces using the Jaeger operator and the "sidecar.jaegertracing.io/inject: jaeger" annotation. However I'm now wanting to use Kiali as well but when I use the Istio sidecar "sidecar.istio.io/inject: "true"" along with the Jaeger sidecar, I'm no longer able to access my routes for my application. 

When I omit the Jaeger sidecar and only use the Istio sidecar, Jaeger does display traces but only the Proxy spans. Am I able to retrieve the traces I'd normally but with only the Istio sidecar, or is there any extra configuration required to be able to use both sidecars?

Thanks.

**Comment by objectiser** [comment]
Hi @AnthonyGurg 

Currently if you want to have spans reported for the inbound/outbound requests (from the Istio/envoy proxy), and also spans from within the application being traced using Jaeger, then it would be necessary to have both the Istio sidecar and a Jaeger tracer/client reporting spans - either to a sidecar Jaeger agent or directly to the collector.

Hopefully in the future, when Istio move to a more OpenTelemetry based approach, it would be possible to have the OpenTelemetry Collector as an agent bundled within the Istio sidecar, and as well as the proxy reporting its own spans, the app would also be able to report its observability data (metrics, logs and traces) to that same sidecar. This would definitely simplify configuration.

**Reply by AnthonyGurg** [reply]
Hi @objectiser, thanks for the response. I actually don't want/care about the spans from the proxy, however I believe the Istio sidecar is required to be able to view traces in Kiali along with metrics and such which is why I wanted both sidecars. Thanks.

**Comment by jpkrohling** [comment]
> when I use the Istio sidecar "sidecar.istio.io/inject: "true"" along with the Jaeger sidecar, I'm no longer able to access my routes for my application.

As a sanity check: are you able to get the BookInfo demo working fine? 

P.S.: if you have an OpenShift subscription (looks like you do), you might want to open a support case instead. We don't have an SLA for support here in the community forum.

**Reply by AnthonyGurg** [reply]
> > when I use the Istio sidecar "sidecar.istio.io/inject: "true"" along with the Jaeger sidecar, I'm no longer able to access my routes for my application.
> 
> As a sanity check: are you able to get the BookInfo demo working fine?
> 
> P.S.: if you have an OpenShift subscription (looks like you do), you might want to open a support case instead. We don't have an SLA for support here in the community forum.

I did test out the Bookinfo demo and it worked completely fine but also didn't require the Jaeger sidecar, only the Istio sidecar. It did show the full spans though. I can open up a support case and see if they're able to help though thank you.

**Reply by jpkrohling** [reply]
Actually, I think bookinfo does not generate spans from the application, only from Istio. A support will certainly be good for you, as you'll have some SLA. It will probably land on us here anyway, so, it probably won't matter much for this specific case.
